generator/internal/descriptor_utils.cc

```diff
@@ -212,18 +212,23 @@ std::string DefaultIdempotencyFromHttpOperation(
   }
 }
 
+std::string ChompByValue(std::string const& s) {
+  return (s[s.size() - 1] == '\n') ? s.substr(0, s.size() - 1) : s;
```

Comment:
This is UB if `s.empty() == true`. How about `!s.empty() && s.back() == '\n' ? ...`

Or [`absl::StripSuffix()`](https://github.com/abseil/abseil-cpp/blob/a4798817e824a5c45689933bc62338f3c7d6961b/absl/strings/strip.h#L82-L83)